### PR TITLE
Fix built-in release version

### PR DESCRIPTION
**Pull request by timoreimann**
In #274, we started to release container images by promoting/retagging the latest `:master` image. However, this did not yield an updated version string baked into the binary; instead, we ended up with the _latest_ version used when we build the latest master.

This change modifies the release Github Action by recompiling and packaging the binary so that the right version is included.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,7 +44,7 @@ endif
 
 all: check-unused test
 
-publish: compile build push clean
+publish: clean compile build push
```
Comment (timoreimann) [inline Makefile:47]: Drive-by improvement: cleaning after publishing didn't really make sense to me.

**Review by MorrisLaw — APPROVED** [review]
